specs/experimental/op-contracts-manager.md

```diff
@@ -18,148 +18,282 @@ of governance approved [contract releases] can be found on the
-specified by the [standard configuration].
+### Interface
 
-### `deploy`
+#### `deploy`
 
-The `deploy` method is the only non-view method in the contract. It is used to
-deploy the full set of L1 contracts required to setup a new OP Stack chain that
-complies with the [standard configuration]. It has the following interface:
+The `deploy` method is used to deploy the full set of L1 contracts required to setup a new OP Stack
+chain that complies with the [standard configuration]. It has the following interface:
 
 ```solidity
-struct Roles {
-  address proxyAdminOwner;
-  address systemConfigOwner;
-  address batcher;
-  address unsafeBlockSigner;
-  address proposer;
-  address challenger;
-}
-
-function deploy(
-  uint256 l2ChainId,
-  uint32 basefeeScalar,
-  uint32 blobBasefeeScalar,
-  Roles roles
-) external returns (SystemConfig)
+/// @notice Deploys a new OP Chain
+/// @param _input DeployInput containing chain specific config information.
+/// @return DeployOutput containing the new addresses.
+function deploy(DeployInput calldata _input) external returns (DeployOutput memory)
 ```
 
 The `l2ChainId` has the following restrictions:
 
 - It must not be equal to 0.
 - It must not be equal to the chain ID of the chain the OP Contracts Manager is
-deployed on.
+  deployed on.
 - It must not be equal to a chain ID that is already present in the
-[ethereum-lists/chains] repository. This is not enforced onchain, but may matter
-for future versions of OP Contracts Manager that handle upgrades.
+  [ethereum-lists/chains] repository. This is not enforced onchain, but may matter
+  for future versions of OP Contracts Manager that handle upgrades.
 
 On success, the following event is emitted:
 
 ```solidity
-event Deployed(uint256 indexed l2ChainId, SystemConfig indexed systemConfig);
+event Deployed(uint256 indexed outputVersion, uint256 indexed l2ChainId, address indexed deployer, bytes deployOutput);
 ```
 
 This method reverts on failure. This occurs when:
 
 - The input `l2ChainId` does not comply with the restrictions above.
 - The resulting configuration is not compliant with the [standard configuration].
 
-### Getter Methods
+### Implementation
 
-The following interface defines the available getter methods:
+#### Batch Inbox Address
+
+The chain's [Batch Inbox] address is computed at deploy time using the recommend approach defined
+in the [standard configuration]. This improves UX by removing an input, and ensures uniqueness of
+the batch inbox addresses.
+
+#### Contract Deployments
+
+All contracts deployed by the OP Contracts Manager are deployed with CREATE2.
+
+For singletons the following salt is used:
 
 ```solidity
-/// @notice The logic address and initializer selector for an implementation contract.
-struct Implementation {
-  address logic; // Address containing the deployed logic contract.
-  bytes4 initializer; // Function selector for the initializer.
+keccak256(abi.encode(l2ChainId, saltMixer))
```

Comment:
The implementation of salts have now been simplied as per this pr: https://github.com/ethereum-optimism/optimism/pull/13273 

Now all salts are computed the _same_. So both proxied and non-proxied contracts hav their salts computed as: 
```solidity
keccak256(abi.encode(_l2ChainId, _saltMixer, _contractName));
```

Reply:
Thanks for following up here! I think that's a great improvement which I've reflected here: ec5563349ed1